test/integration_test.py

```diff
@@ -1104,20 +1104,25 @@ def test_can_bundle_configured_catalogs(self):
 
     @mock.patch.dict(os.environ,
                      azul_canned_sources=f'{canned_repo}:')
-    @mock.patch.object(azul.Config, 'catalogs', new_callable=PropertyMock)
-    # Must be mocked independently due to caching
-    @mock.patch.object(azul.Config, 'default_catalog', new_callable=PropertyMock)
-    def test_can_bundle_canned_repository(self, mock_default_catalog, mock_catalogs):
+    def test_can_bundle_canned_repository(self):
         mock_catalog_name = 'testcanned'
         mock_catalog = config.Catalog(name=mock_catalog_name,
                                       atlas='hca',
                                       plugins={
                                           'metadata': 'hca',
                                           'repository': 'canned'
                                       })
-        mock_default_catalog.return_value = mock_catalog_name
-        mock_catalogs.return_value = {mock_catalog_name: mock_catalog}
-        self._test_catalog(mock_catalog)
+
+        with mock.patch.object(azul.Config,
+                               'catalogs',
+                               new=PropertyMock(return_value={
+                                   mock_catalog_name: mock_catalog
+                               })):
+            # Reset cached property to reflect mock
+            azul.Config.default_catalog.fdel(config)
+            self._test_catalog(mock_catalog)
+        # Reset cache again so mock doesn't leak into other tests
+        azul.Config.default_catalog.fdel(config)
```

Comment:
Hmm, I didn't realize this is necessary. Technically, it would need to go into a `finally` block which would probably be the same amount of code as a patch. The only advantage of this current approach over patching the cached property is that it doesn't need to know the value of the cached property, only that it is derived and cached.

Reply:
I'll push a commit for stashing cached properties.

Reply:
This is a really neat addition, but since `config.default_property` [was changed](https://github.com/DataBiosphere/azul/commit/8e9d8206891e309ef0ce29dc4f8d76595115ecdf#diff-a5ab528d23a14c23dd1439c679fe8f2af5e5bcab793aee95451aeefe4f7b86dfL544) from a `cached_property` to a regular ol' `property` in the meantime, there's no longer any need for this here.